modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1579,7 +1590,11 @@ private Map<String, Serializable> _toObjectValues(
 				continue;
 			}
 
-			if ((value == null) && !objectField.isRequired()) {
+			if ((value == null) &&
+				(!objectField.isRequired() ||
+				 (_getStatusCode(objectEntry.getStatus()) ==
+					 WorkflowConstants.STATUS_DRAFT))) {
```

Comment:
The current behavior is: if the value is null and the field is not required, it does not get added to the 'values' Map (so nothing is persisted for that field). If the field is required, its key is added to the 'values' Map and an exception is thrown here (https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L4154).
Because we don't check for required fields when saving as draft, in that case we would end up saving "null" literal in DB.

Reply:
Perfect. Thanks